provenance.jq

```diff
@@ -0,0 +1,75 @@
+# input: "build" object with platform and image digest
+#   $github: "github" context; CONTAINS SENSITIVE INFORMATION (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context)
+#   $runner: "runner" context; https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#runner-context
+#   $digest: the OCI image digest for the just-built image (normally in .build.resolved.annotations["org.opencontainers.image.ref.name"] but only post-push/regeneration and we haven't pushed yet)
+#
+# output: in-toto provenance statement (https://slsa.dev/spec/v1.0/provenance)
+#   see also: https://github.com/actions/buildtypes/tree/main/workflow/v1
+def github_actions_provenance($github; $runner; $digest):
+	if $github.event_name != "workflow_dispatch" then error("error: '\($github.event_name)' is not a supported event type for provenance generation") else
```

Comment:
Oh, I forgot to mention this in my commit message, so I'll highlight it with a comment instead -- I was looking at https://github.com/actions/buildtypes/tree/main/workflow/v1 and the list of events it supports, and realized our provenance has even tighter assumptions than theirs (especially for `$github.event.inputs`) so made anything else an error accordingly just to be safe. :+1:

Reply:
yes, I noticed this and definitely like the addition 👍